binding/python/docs/TUTORIAL.md

```diff
@@ -0,0 +1,144 @@
+```
+self.W = theano.shared(
+    value=numpy.zeros(
+        (n_in, n_out),
+        dtype=theano.config.floatX
+    ),
+    name='W',
+    borrow=True
+)
+```
+
+If you want to use multiverso, you can modify them like this.
+```
+from multiverso.theano_ext import sharedvar
+# Only the master worker can initialize the shared value. The initial value from
+# other processes will be ignored
+W = sharedvar.mv_shared(
+    value=numpy.zeros(
+        (n_in, n_out),
+        dtype=theano.config.floatX
+    ),
+    name='W',
+    borrow=True
+)
+
+# build the model
+
+# train the modle
+
+# When you are ready to add the delta of the variable to parameter server and sync the latest value, you can run this function
+W.mv_sync()
+
+
+# If you want to sync all variables created by `sharedvar.mv_shared`, you can use this function
+sharedvar.sync_all_mv_shared_vars()
+```
+
+`mv_shared` is just a wrapper of `theano.shared`. It acts same as `theano.shared`, but it give you much more convenient interface to sync values.
+
+If your program will run in multi-process and you want to initialize your parameters, you should initialize your shared variables with the value you want only in one process and initialize them with zero in other processes.
+
+
+If you don't use shared variables to store and update the parameters, you can still use the `add` and `get` functions to sync parameters.
+
+
+# How to use multiverso in lasagne
+First, make sure you have understood the last section.
+
+Lasagne provides many functions to help you build models more easily. Multiverso python binding provides a manager to make managing and synchronizing the parameters in Lasagne more easily.
+
+You can write code like this to manage your parameters.
+```
+from multiverso.theano_ext.lasagne_ext import param_manager
+
+network = build_model()  # build_model is a function you implement to build model
+
+# The MVNetParamManager will initialize the parameters and sync them with
+# parameter server
+mvnpm = param_manager.MVNetParamManager(network)
+
+# training the model
+
+# When you are ready to add the delta of the variable in this model to the parameter server and get the latest value, you can run this function
+mvnpm.sync_all_param()
+```
+
+
+# Run your multiverso program with multi-process
+Here is an example of running logistic regression with multi-process.
+```
+mpirun -np 4 python ./examples/theano/logistic_regression.py
+```
+
+
+# How to use multi-GPU in theano with multiverso
+You need multiple GPUs in your server and have installed the [CUDA backend](http://deeplearning.net/software/theano/tutorial/using_gpu.html#cuda).
+
+First, make sure you have read [this section](http://deeplearning.net/software/theano/install.html#using-the-gpu) and understand how to configure which GPU will be used.
+
+With multiverso, your program will run in multiple processes.
+
+Here is an example to make different GPUs used in different processes.
```

Comment:
"make different process use different GPUs" or "have different GPUs used in different processes".